evmrpc/info.go

```diff
@@ -143,7 +143,7 @@ func (i *InfoAPI) FeeHistory(ctx context.Context, blockCount math.HexOrDecimal64
 		}
 		result.BaseFee = append(result.BaseFee, (*hexutil.Big)(baseFee))
 		height := blockNum
-		block, err := i.tmClient.Block(ctx, &height)
+		block, err := blockWithRetry(ctx, i.tmClient, &height)
```

Comment:
i'm a bit concerned about the perf impact of always retry - for cases like this we know if the block isn't there on the first try then it won't be there even if we retry.

I think we can pass a `retryAttempts` arg to `blockWithRetry` and set it to 0 for cases we don't really need retries, if the goal here is simply to have unified error handling

Reply:
Good call, this should be fixed now, please take another look